x-pack/plugins/observability_solution/infra/server/routes/custom_dashboards/save_custom_dashboard.ts

```diff
@@ -37,29 +39,36 @@ export function initSaveCustomDashboardRoute(framework: KibanaFramework) {
 
       await checkCustomDashboardsEnabled(uiSettingsClient);
 
-      const payload: InfraSaveCustomDashboardsRequestPayload = request.body;
-      const customDashboards = await findCustomDashboard(payload.assetType, savedObjectsClient);
+      const payload = {
+        ...request.body,
+      };
 
-      if (customDashboards.total === 0) {
-        const savedCustomDashboard = await savedObjectsClient.create<InfraCustomDashboard>(
-          INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE,
-          payload
-        );
+      const { assetType } = request.params;
 
-        return response.ok({
-          body: InfraSaveCustomDashboardsResponseBodyRT.encode(savedCustomDashboard.attributes),
+      const customDashboards = await findCustomDashboard(assetType, savedObjectsClient);
+
+      const dashboardExist = customDashboards.find(
+        (customDashboard) =>
+          customDashboard.dashboardSavedObjectId === payload.dashboardSavedObjectId
+      );
+
+      if (dashboardExist) {
+        return response.badRequest({
+          body: `Dashboard with id ${payload.dashboardSavedObjectId} has already been linked to ${assetType}`,
         });
       }
 
-      const savedCustomDashboard = await savedObjectsClient.update<InfraCustomDashboard>(
+      const savedCustomDashboard = await savedObjectsClient.create<InfraCustomDashboard>(
         INFRA_CUSTOM_DASHBOARDS_SAVED_OBJECT_TYPE,
-        customDashboards.saved_objects[0].id,
-        payload
+        {
+          assetType,
+          ...payload,
```

Comment:
last nit:

```suggestion
          ...request.body,
```

Reply:
Here we use `dashboardSavedObjectId` from the `payload`. So I changed it to  
`const { dashboardSavedObjectId } = request.body;` and then used the `...request.body,` here